### DISCUSSION TITLE
v11 WebAssembly support

### DISCUSSION THREAD

**Discussion by ScottShingler** [AvaloniaUI/Avalonia#13330] [Q&A] [answered]
I'm confused as to what the state of WASM is in Avalonia v11.

On the one hand, the [Platforms](https://avaloniaui.net/Platforms#wasm) page states:

> With our comprehensive WebAssembly support, your application can be deployed directly to a web without altering a single line of code.

On the other hand, the [documentation](https://docs.avaloniaui.net/docs/next/guides/platforms/how-to-use-web-assembly) states:

> It is currently very early days and not ready for production, however if you want to test this exciting new feature please take the following steps.

Maybe I'm misinterpreting something, but these statements seem to be at odds with each other. What's the state of WebAssembly support in v11?

**Comment by robloo** [comment]
"Marketing" vs. "Engineering" speech I would guess. That said, where support is now is functional and only going to get better with .NET 8. I'll let the core team expand on that if they wish.

**Comment by maxkatz6** [comment, marked as answer]
"not ready for production" was kept there since 0.10 days. It's time to remove it.

Saying it, I would say that WASM (WASM in general, not Avalonia WASM) is not the best fit for any production application. Limited threading (non-existent pre-.NET 7), sandboxing, and performance bottlenecks make it more limited for just anybody.

**Reply by maxkatz6** [reply]
Updated that article, should be more up to date now.

**Reply by kyuranger** [reply]
BTW,will these problems be solved for .NET8?

**Reply by maxkatz6** [reply]
These are no problems, but browser limitations. Only WASM multithreading will be added in .NET 8, but even then it's going to be very limited. Unfortunately, developers need to learn how to write apps with these limitations.
You can check this up-to-date topic about multithreading in .NET 8 WASM: https://github.com/dotnet/designs/blob/346cd7db900a917dde7ca67fbcd6785a73935709/accepted/2023/wasm-browser-threads.md
